backend/src/v4/handler/alluxio.js

```diff
@@ -0,0 +1,90 @@
+/**
+ *  Copyright (C) 2020 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+"use strict";
+
+const config = require("../config.js");
+const path = require("path");
+const ResponseCodes = require("../response_codes");
+const nodeuuid = require("uuid").v1;
+const farmhash = require("farmhash");
+const AlluxioClient = require("../models/alluxioClient");
+const slash = require("slash");
+
+const generateFoldernames = (fileName, dirLevels) => {
+	if (dirLevels < 1) {
+		return "";
+	}
+	const folders = [];
+	const minChunkLen = 4;
+	const nameChunkLen = Math.max(fileName.length / dirLevels, minChunkLen);
+
+	for(let i = 0 ; i < dirLevels; i++) {
+		const chunkStart = (i * nameChunkLen) % fileName.length;
+		const fileNameHash = farmhash.fingerprint32(fileName.substr(chunkStart,nameChunkLen) + Math.random());
+		folders.push(fileNameHash & 255);
+	}
+	return folders.join("/");
+};
+
+class AlluxioHandler {
+	constructor() {
+		if (config.alluxio) {
+			const {hostname, port, levels } = config.alluxio;
+			this.client = new AlluxioClient(`${hostname}:${port}`);
```

Comment:
can we test the connection and break it if it fails? We do this for every service we launch